File: frontend/src/types/geoAnalytics.ts
export interface SolarPotentialZone {
  id: string;
  name: string;
  state: string;
  irradiance_kwh_m2: number;
  land_suitability_score: number;
  area_sq_km: number;
}

export interface GridInfrastructure {
  id: string;
  name: string;
  infrastructure_type: string;
  capacity_mva: number | null;
  congestion_level: number | null;
}

export interface DisasterRiskZone {
  id: string;
  risk_type: string;
  severity: number;
  state: string;
  description: string;
}

// ---------------------------------------------------------------------------
// Groundwater Resource Assessment (SolarWindAssessment integration)
// ---------------------------------------------------------------------------

export interface GroundwaterFeatureCollection {
  type: "FeatureCollection";
  features: Array<{
    type: "Feature";
    geometry: object | null;
    properties: {
      id: string;
      gml_id: string;
      objectid: number | null;
      block: string;
      tehsil: string;
      district: string;
      gwr_2011_2: number | null;
      code: number | null;
      classification: string;
      net_annual_gw_availability: number | null;
      annual_gw_draft_irrigation: number | null;
      stage_of_gw_development: number | null;
      annual_gw_draft_domestic_industrial: number | null;
      annual_gw_draft_total: number | null;
      annual_replenishable_gw_total: number | null;
      state_name: string;
      natural_discharge_non_monsoon: number | null;
      st_area_shape: number | null;
      st_length_shape: number | null;
    };
  }>;
}

// ---------------------------------------------------------------------------
// Google Service Credential (SolarWindAssessment integration)
// ---------------------------------------------------------------------------

export interface GoogleServiceCredential {
  id: string;
  name: string;
  credential_type: string;
  project_id: string;
  private_key_id: string;
  client_email: string;
  client_id: string;
  auth_uri: string;
  token_uri: string;
  auth_provider_x509_cert_url: string;
  client_x509_cert_url: string;
  is_active: boolean;
}

File: backend/app/domains/geo_analytics/schemas/geo_analytics.py
from uuid import UUID

from pydantic import BaseModel


class SolarPotentialZoneRead(BaseModel):
    id: UUID
    name: str
    state: str
    irradiance_kwh_m2: float
    land_suitability_score: float
    area_sq_km: float


class GridInfrastructureRead(BaseModel):
    id: UUID
    name: str
    infrastructure_type: str
    capacity_mva: float | None = None
    congestion_level: int | None = None


class DisasterRiskZoneRead(BaseModel):
    id: UUID
    risk_type: str
    severity: int
    state: str
    description: str


class SolarPotentialQuery(BaseModel):
    state: str | None = None
    min_irradiance: float | None = None
    min_suitability: float | None = None


# ---------------------------------------------------------------------------
# Groundwater Resource Assessment (SolarWindAssessment integration)
# ---------------------------------------------------------------------------

class GroundwaterResourceRead(BaseModel):
    id: UUID
    gml_id: str
    objectid: int | None = None
    block: str
    tehsil: str
    district: str
    gwr_2011_2: float | None = None
    code: float | None = None
    classification: str
    net_annual_gw_availability: float | None = None
    annual_gw_draft_irrigation: float | None = None
    stage_of_gw_development: float | None = None
    annual_gw_draft_domestic_industrial: float | None = None
    annual_gw_draft_total: float | None = None
    annual_replenishable_gw_total: float | None = None
    state_name: str
    natural_discharge_non_monsoon: float | None = None
    st_area_shape: float | None = None
    st_length_shape: float | None = None
    # GeoJSON geometry as a plain dict so it can be serialised directly to the frontend
    geometry: dict | None = None


class GroundwaterResourceFeatureCollection(BaseModel):
    """GeoJSON FeatureCollection wrapper returned by the API."""

    type: str = "FeatureCollection"
    features: list[dict]


# ---------------------------------------------------------------------------
# Google Service Credentials (SolarWindAssessment integration)
# ---------------------------------------------------------------------------

class GoogleServiceCredentialRead(BaseModel):
    """Public view – private_key is intentionally omitted."""

    id: UUID
    name: str
    credential_type: str
    project_id: str
    private_key_id: str
    client_email: str
    client_id: str
    auth_uri: str
    token_uri: str
    auth_provider_x509_cert_url: str
    client_x509_cert_url: str
    is_active: bool


class GoogleServiceCredentialCreate(BaseModel):
    name: str
    credential_type: str = "service_account"
    project_id: str
    private_key_id: str
    private_key: str
    client_email: str
    client_id: str
    auth_uri: str = "https://accounts.google.com/o/oauth2/auth"
    token_uri: str = "https://oauth2.googleapis.com/token"
    auth_provider_x509_cert_url: str = "https://www.googleapis.com/oauth2/v1/certs"
    client_x509_cert_url: str = ""
